src/workerd/jsg/jsvalue.h

```diff
@@ -0,0 +1,819 @@
+#pragma once
+
+#include "jsg.h"
+#include <v8.h>
+
+namespace workerd::jsg {
+
+// Disables heap allocation while still allowing for placement new.
+// This enables a type T to disable, for instance, kj::heap<T> while
+// still being usable with kj::Maybe and jsg::Optional.
+#define JSG_REQUIRE_STACK_ALLOCATION                                     \
```

Comment:
The thing I like about this macro is that it does end up with a compile-time error if someone tries to do something silly like `kj::heap<JsValue>` (as unlikely as it may be). I recognize it doesn't address the full requirement but it does seem to me to be a reasonable guard rail. The code is relying on `requireOnStack()` to fully enforce the requirement. I could be convinced that the compile error is not worth it.

Reply:
Right. I think the compile error is not worth it because I think it's extremely rare to allocate a bare value like that on the heap in the first place, so it'll probably never catch anyone, and even if someone does try to do that, the runtime error will catch it. Meanwhile this code has to directly reference KJ private symbols in order to re-enable `kj::ctor`. Seems not worth it to me.

Reply:
ok, fair enough. removed!